code/src/components/GamePage.js

```diff
@@ -0,0 +1,183 @@
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import { Direction } from "./Direction";
+import { EndPage } from "./EndPage";
+import map from "images/map.png";
+import Lottie from "react-lottie";
+import animationData from "../lotties/loading";
+import animationDataShip from "../lotties/ship";
+import animationDataWater from "../lotties/water";
+
+export const GamePage = () => {
+  const labyrinth = useSelector((store) => store.labyrinth.location);
+  const loading = useSelector((store) => store.ui.loading);
+  const [setDirection] = useState("");
+
+  const defaultOptions = {
+    loop: true,
+    autoplay: true,
+    animationData: animationData,
+    rendererSettings: {
+      preserveAspectRatio: "xMidYMid slice",
+    },
+  };
+
+  const defaultOptionsShip = {
+    loop: true,
+    autoplay: true,
+    animationData: animationDataShip,
+    rendererSettings: {
+      preserveAspectRatio: "xMidYMid slice",
+    },
+  };
+
+  const defaultOptionsWater = {
+    loop: true,
+    autoplay: true,
+    animationData: animationDataWater,
+    rendererSettings: {
+      preserveAspectRatio: "xMidYMid slice",
+    },
+  };
+
+  const mapTilesArray = [
+    [false, false],
+    [false, false],
+    [false, false],
+    [false, false],
+  ];
+
+  const createMapTiles = () => {
+    let mapTiles = [];
+    // for-loop over rows (in backward direction so that first is at bottom)
+    for (let i = 3; i > -1; i--) {
+      mapTiles.push(<div className="black map-tile"></div>);
+      // for-loop for columns
+      for (let j = 0; j < 2; j++) {
+        if (mapTilesArray[i][j]) {
+          mapTiles.push(
+            <div className="white">
+              <>
+                <Lottie options={defaultOptionsShip} width={100} height={100} />
+              </>
+            </div>
+          );
+        } else {
+          mapTiles.push(<div></div>);
+        }
+      }
+
+      mapTiles.push(<div className="black map-tile"></div>);
+    }
+    return mapTiles;
+  };
+
+  // sets the current map tile to true, which triggers a class name change
+  const setMapTile = (coords) => {
+    const ordinates = coords.split(",");
+    mapTilesArray[Number(ordinates[1])][Number(ordinates[0])] = true;
+  };
+
```

Comment:
Congrats to make this work! We don't really understand it though, because it's quite advanced 🤣